src/transformers/models/vocos/modeling_vocos.py

```diff
@@ -0,0 +1,302 @@
+# coding=utf-8
+# Copyright 2025 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Optional, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ...audio_utils import istft
+from ...modeling_utils import PreTrainedModel
+from ...processing_utils import Unpack
+from ...utils import ModelOutput, TransformersKwargs, auto_docstring, can_return_tuple
+from .configuration_vocos import VocosConfig
+
+
+@dataclass
+class VocosOutput(ModelOutput):
+    """
+    Args:
+        audio (`torch.FloatTensor` of shape `(batch_size, time)`):
+            Reconstructed audio waveform.
+    """
+
+    audio: torch.FloatTensor
+
+
+class VocosAdaptiveLayerNorm(nn.Module):
+    """
+    Weight and bias parameters come from a lookup table based on the target bandwidth.
+    """
+
+    def __init__(self, config: VocosConfig):
+        super().__init__()
+        self.eps = config.layer_norm_eps
+        self.hidden_size = config.hidden_size
+        adanorm_num_embeddings = len(config.bandwidths)
+        self.weight = nn.Parameter(torch.ones(adanorm_num_embeddings, config.hidden_size))
+        self.bias = nn.Parameter(torch.zeros(adanorm_num_embeddings, config.hidden_size))
+
+    def forward(self, hidden_states: torch.Tensor, cond_embedding_id: torch.LongTensor):
+        hidden_states = F.layer_norm(hidden_states, (self.hidden_size,), weight=None, bias=None, eps=self.eps)
+        return hidden_states * self.weight[cond_embedding_id].unsqueeze(0) + self.bias[cond_embedding_id].unsqueeze(0)
+
+
+class VocosConvNeXtBlock(nn.Module):
+    """ConvNeXt block adapted for 1D convolutions in the Vocos architecture."""
+
+    def __init__(self, config: VocosConfig):
+        super().__init__()
+        self.dwconv = nn.Conv1d(
+            config.hidden_size,
+            config.hidden_size,
+            kernel_size=config.kernel_size,
+            padding=config.padding,
+            groups=config.hidden_size,
+        )
+        if config.use_adaptive_norm:
+            self.norm = VocosAdaptiveLayerNorm(config)
+        else:
+            self.norm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.pwconv1 = nn.Linear(config.hidden_size, config.intermediate_size)
+        self.act = nn.GELU()
```

Comment:
Nit but let's use Llama convention.
```suggestion
        self.act_fn = ACT2FN["gelu"]
```